### PR TITLE
MAGECLOUD-4407: Remove Deprecated Variables from ECE-Tools

### PR DESCRIPTION
**Pull request by BaDos**
### Fixed Issues (if relevant)

1. https://magento2.atlassian.net/browse/MAGECLOUD-4407
2. https://magento2.atlassian.net/browse/MAGECLOUD-4520

### Manual testing scenarios

1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by oshmyheliuk — DISMISSED** [review]

File: scenario/build/generate.xml [hunk 1]
```diff
@@ -13,7 +13,6 @@
                 <item name="300" xsi:type="array">
                     <item name="config-file-exists" xsi:type="object" priority="100">Magento\MagentoCloud\Config\Validator\Build\ConfigFileExists</item>
                     <item name="deprecated-ini" xsi:type="object" priority="200">Magento\MagentoCloud\Config\Validator\Build\UnsupportedBuildOptionsIni</item>
-                    <item name="stage-config-deprecated-variables" xsi:type="object" priority="300">Magento\MagentoCloud\Config\Validator\Build\StageConfigDeprecatedVariables</item>
```
Comment (oshmyheliuk) [inline scenario/build/generate.xml:16]: Please adjust the priority for other steps.

File: src/Config/Validator/Deploy/DeprecatedVariables.php [hunk 2]
```diff
@@ -75,44 +65,6 @@ public function validate(): Validator\ResultInterface
...
-            && $config[DeployInterface::VAR_STATIC_CONTENT_SYMLINK] === false
-        ) {
-            $errors[] = sprintf(
-                'The %s variable is deprecated and its behavior will not be supported in the future.',
-                DeployInterface::VAR_STATIC_CONTENT_SYMLINK
-            );
-        }
-
         if ($errors) {
             return $this->resultFactory->error(
```
Comment (oshmyheliuk) [inline src/Config/Validator/Deploy/DeprecatedVariables.php:60]: Now you can return `error` in the previous condition, there is no need to keep `$errors` array for the one possible error.

**Review by shiftedreality — DISMISSED** [review]

File: src/Config/Validator/Deploy/DeprecatedVariables.php (hunk 2 above)
Comment (BaDos) [reply src/Config/Validator/Deploy/DeprecatedVariables.php:60]: Fixed

File: scenario/build/generate.xml (hunk 1 above)
Comment (BaDos) [reply scenario/build/generate.xml:16]: Done

**Review by oshmyheliuk — DISMISSED** [review]

**Review by shiftedreality — DISMISSED** [review]

**Comment by mveeramneni** [comment]
Will run Magecat tests on this PR

**Review by shiftedreality — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by mveeramneni** [comment]
QA Approved

**Comment by ghost** [comment]
Hi @BaDos, thank you for your contribution! </br>Please, complete [Contribution Survey](https://opensource.magento.com/survey/contribution/magento+ece-tools+629), it will take less than a minute. </br>Your feedback will help us to improve contribution process.

**Pull request merged**